libethashseal/Ethash.cpp

```diff
@@ -152,8 +152,13 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 	{
 		if (_bi.number() >= chainParams().u256Param("EIP158ForkBlock"))
 		{
-			int chainID(chainParams().u256Param("chainID"));
-			_t.checkChainId(chainID);
+			if (_bi.number() >= chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
+				_t.checkChainId(0);
```

Comment:
if you have zero signature, you can extract the chain Id
`chain_id = signature.v`
like `_t.checkChainId(_t.signature.v);`